shared/lsp/registration.ts

```diff
@@ -389,33 +389,47 @@ export async function register({
         }
     }
 
+    function addRoot(root: sourcegraph.WorkspaceRoot): void {
+        const connectionPromise = (async () => {
+            try {
+                const serverRootUri = clientToServerURI(
+                    new URL(root.uri.toString())
```

Comment:
Isn't `root.uri` already a `URL` object?
```suggestion
                    root.uri
```

Reply:
Ah I think this is left over from a transition period where we had to support both strings and URL objects

Reply:
Updated in a few other places as well.